qiskit/providers/fake_provider/fake_backend.py

```diff
@@ -73,6 +73,18 @@ class FakeBackendV2(BackendV2):
 
     def __init__(self):
         """FakeBackendV2 initializer."""
+        # This is a deprecation warning for the subclasses.
+        # FakeBackendV2 is not deprecated.
+        warnings.warn(
```

Comment:
This will still fire for direct use of `FakeBackendV2`, won't it? Also, is there a reason not to deprecate `FakeBackendV2`?

Reply:
So here I'm abusing a bit the fact that `FakeBackendV2` cannot be used directly (it raises an error for the missing config files) to not have to add individual deprecation warnings (and inits) to each fake backend class. I didn't pay much attention to the case where someone might try to instantiate it directly, but now that you mention it, I could move the warning to the end of the init to make sure that the error is raised before the deprecation warning would be raised. I did think that this might also trigger deprecation warnings on anyone who has a custom `FakeBackendV2` subclass, and tried to adjust the message to exclude those cases. Maybe I could add something like "if you are using a custom fake backend implementation, you don't need to take any action". 

I thought it might be too rushed to deprecate the class because, at the moment, we don't have any other mechanism in qiskit to create backends from config files. The provider (well, `qiskit-ibm-runtime`) has its own copy of these classes so in that sense there would be no issues if we deprecated it, but I took the conservative path here.

Reply:
That makes sense, though I believe the config files we expect are IBM-specific, right? IMO the easiest path is to just deprecate this entire class. If someone wants to read settings from their own config file format, they'll need to rewrite it anyway, and if they want to use IBM-format files, they should install the IBM runtime package 😃.

Wondering if @mtreinish has any additional insights?

Reply:
Ok, I think that's reasonable. Added the new deprecation in [1508e8f](https://github.com/Qiskit/qiskit/pull/11381/commits/1508e8f2d901ca28115dcd897965f6dedf2a65b3).